### ISSUE TITLE
[Feature] Selective Stickiness

### ISSUE THREAD

**Issue by supersaiyansubtlety** [williambl/piston-control#4]
Suggestion: allow selective stickiness (like slime/honey and modded colored slime) via tags as follows:
- look in `/data/pistoncontrol/tags/blocks/selectively_sticky/`
- each tag in that directory defines a 'channel' of stickiness that the blocks in the tag belong to
- sticky blocks don't stick to other sticky blocks if they don't share a tag (or untagged stick to everything)
So blocks that appear in more than one `selectively_sticky` tag would stick to blocks in both of those tags, but not other tags. 

This would, for example, allow players to re-create modded colored slime-like behavior where colors of a block stick to colors that are one step away on the color wheel (so orange would stick to yellow and red, but not blue) (they could apply it to terracotta or concrete or something). 

If you're accepting PRs I'd be willing to work in this feature.

**Comment by williambl** [comment]
This is a great feature request, thanks for making it. I'm a little busy with other projects right now, but I'd be happy to merge this if you make a PR. If you don't have the time to do the whole feature and just start on it, I'd be happy to pick up where you left off, too.

**Comment by supersaiyansubtlety** [comment]
Great, I'll start looking into it.

**Comment by supersaiyansubtlety** [comment]
Update: I've got the selectively sticky logic working as far as I can tell. I'm still having trouble getting all block tags in `/data/pistoncontrol/tags/blocks/selectively_sticky/`. Currently I've just tested using tags from `BlockTags`: `PLANKS` and `LOGS`, and made `oak_planks` and `oak_log` sticky. This makes the oak_planks/oak_log act like slime/honey. 

If you create a new branch for this feature development, I'd like to make a PR with my current progress.
